src/dynamic_honey_badger.rs

```diff
@@ -1,3 +1,45 @@
+//! # Dynamic Honey Badger
+//!
+//! Like Honey Badger, this protocol allows a network of `N` nodes with at most `f` faulty ones,
+//! where `3 * f < N`, to input "transactions" - any kind of data -, and to agree on a sequence of
+//! _batches_ of transactions. The protocol proceeds in _epochs_, starting at number 0, and outputs
+//! one batch in each epoch. It never terminates: It handles a continuous stream of incoming
+//! transactions and keeps producing new batches from them. All correct nodes will output the same
+//! batch for each epoch.
+//!
+//! Unlike Honey Badger, this algorithm allows dynamically adding new full peers from the pool of
+//! observer nodes, and turning full peers back into observers. As a signal to initiate that
+//! process, it defines a special `Change` input variant, which contains either a vote
+//! `Add(node_id, public_key)`, to add a new full peer, or `Remove(node_id)` to remove it. Each
+//! full peer can have at most one active vote, and casting another vote revokes the previous one.
```

Comment:
No, any number of votes is fine, and it's always the last one — i.e. the last one that came _out_ of Honey Badger in a batch — that counts. Every vote therefore revokes the previous one and replaces it.

I think we should allow any number of votes per node. It can always happen that e.g. the validators voted to add a node but then DKG never completes (e.g. because the node went offline). Then they should be able to abort the addition, and vote for something else (e.g. to add a different node).